lib/wasi/src/syscalls/mod.rs

```diff
@@ -2829,10 +2824,15 @@ pub fn path_symlink<M: MemorySize>(
         wasi_try!(state
             .fs
             .get_parent_inode_at_path(inodes.deref_mut(), fd, old_path_path, true));
-    let depth = wasi_try!(state
+    let depth = state
         .fs
-        .path_depth_from_fd(inodes.deref(), fd, source_inode))
-        - 1;
+        .path_depth_from_fd(inodes.deref(), fd, source_inode);
+
+    // depth == -1 means folder is not relative
```

Comment:
Is it worth mentioning the ticket number or failing test that caused this change?

```suggestion
    // depth == -1 means folder is not relative. See issue #3233.
```